yogstation/code/datums/weather/nuclear_impact.dm

```diff
@@ -0,0 +1,122 @@
+/datum/weather/nuclear
+	name = "nuclear detonation"
+	desc = "A tactical nuclear detonation has been detected in station airspace."
+	telegraph_duration = 280
+	telegraph_message = "<span class='boldwarning'>A loud alarm blares in your ears as the sky turns orange, SEEK SHELTER!</span>"
+	weather_message = "<span class='userdanger'><i>The air around you grows unbearably hot.</i></span>"
+	weather_overlay = "nuclear_heatwave"
+	end_overlay = "light_ash"
+	telegraph_overlay = "nuclear_light_heatwave"
+	weather_duration_lower = 200
+	weather_duration_upper = 400
+	weather_color = null
+	telegraph_sound = 'yogstation/sound/effects/nuclearblast.ogg'
+	weather_sound = 'yogstation/sound/effects/nuclearwind.ogg'
+	end_duration = 300
+	area_type = /area
+	protected_areas = list(/area/maintenance, /area/ai_monitored/turret_protected/ai_upload, /area/ai_monitored/turret_protected/ai_upload_foyer,
+	/area/ai_monitored/turret_protected/ai, /area/storage/emergency/starboard, /area/storage/emergency/port, /area/shuttle)
+	target_trait = ZTRAIT_STATION
+	end_message = "<span class='notice'>The air around you is filled with ash, it's hard to breathe.</span>"
+	immunity_type = null //Are you immune to nukes?
+	end_sound = 'yogstation/sound/effects/nuclearallclear.ogg'
+
+
+/datum/weather/nuclear/update_areas()
+	for(var/V in impacted_areas)
+		var/area/N = V
+		N.layer = overlay_layer
+		N.icon = 'yogstation/icons/effects/weather_effects.dmi'
+		N.color = weather_color
+		switch(stage)
+			if(STARTUP_STAGE)
+				N.icon_state = telegraph_overlay
+			if(MAIN_STAGE)
+				N.icon_state = weather_overlay
+			if(WIND_DOWN_STAGE)
+				N.icon_state = end_overlay
+			if(END_STAGE)
+				N.color = null
+				N.icon_state = ""
+				N.icon = 'icons/turf/areas.dmi'
+				N.layer = AREA_LAYER //Just default back to normal area stuff since I assume setting a var is faster than initial
+				N.set_opacity(FALSE)
+
+/datum/weather/nuclear/weather_act(atom/movable/M)
+	if(isliving(M))
+		if(ishuman(M))
```

Comment:
Why the ``isliving(M)`` check here?

Reply:
i think we need an `ismob(M)` here too to be honest